auto_remediate/security_hub_rules.py

```diff
@@ -75,6 +109,248 @@ def access_keys_rotated(self, resource_id):
             self.logging.error(sys.exc_info()[1])
             return False
 
+    def cloud_trail_cloud_watch_logs_enabled(self, resource_id):
+        """Adds CloudWatch Log Group to CloudTrail logs
+        
+        Arguments:
+            resource_id {string} -- CloudTrail name
+        
+        Returns:
+            boolean -- True if remediation was successful
+        """
+        cloudwatch_log_group_name = f"/aws/cloudtrail/{resource_id}"
+
+        # create CloudWatch Log Group
+        try:
+            self.client_logs.create_log_group(logGroupName=cloudwatch_log_group_name)
+            self.logging.info(
+                f"Created new CloudWatch Log Group '{cloudwatch_log_group_name}'."
+            )
+        except:
+            self.logging.error(
+                f"Could not create new CloudWatch Log Group '{cloudwatch_log_group_name}'."
+            )
+            self.logging.error(sys.exc_info()[1])
+            return False
+        else:
+            try:
+                response = self.client_logs.describe_log_groups(
+                    logGroupNamePrefix=cloudwatch_log_group_name
+                )
+            except:
+                self.logging.error(
+                    f"Could not describe CloudWatch Log Group '{cloudwatch_log_group_name}'."
+                )
+                self.logging.error(sys.exc_info()[1])
+                self.delete_log_group(cloudwatch_log_group_name)
+                return False
+            else:
+                cloudwatch_log_group_arn = response.get("logGroups")[0].get("arn")
+                self.logging.info(
+                    f"Retrieved ARN '{cloudwatch_log_group_arn}' for CloudWatch Log Group '{cloudwatch_log_group_name}'."
+                )
+
+        # get trust relationship
+        try:
+            trust_relationship_file = "auto_remediate/data/cloud_trail_cloud_watch_logs_enabled_trust_relationship.json"
+            with open(trust_relationship_file) as file:
```

Comment:
```python3
with open(file, "r") as file:
    trust = file.read()
```
specify open mode and no need to convert to str

Reply:
61e47e0